Automattic/wp-calypso
# State: Introduce site user type data layer and actions

#### Changes proposed in this Pull Request

* Introduce actions and data layer for saving Jetpack site user type. This is used during the Jetpack NUX site "ownership" step.
Currently we are using 'builder' and 'owner' roles (see https://github.com/Automattic/wp-calypso/pull/31297 for the UI).
#### Testing instructions

* Use this branch 
* Apply D25062-code and sandbox WP.com API
* Load `http://calypso.localhost:3000/jetpack/connect`
* For an id of a WP.com site, fire
```
dispatch( { type: 'JETPACK_CONNECT_SAVE_SITE_USER_TYPE', siteId: id, siteUserType: 'builder' } )
```
from the console.
* Verify you can find `site_user_type`option with the 'builder' value in the `/wp-admin/options.php`.
